src/notation/internal/notationundostack.cpp

```diff
@@ -225,7 +226,7 @@ const muse::TranslatableString NotationUndoStack::undoRedoActionNameAtIdx(size_t
         return {};
     }
 
-    if (auto action = undoStack()->atIndex(idx)) {
+    if (auto action = undoStack()->lastAtIndex(idx)) {
```

Comment:
Feels a little odd to call a method with the name `undoRedoActionNameAtIdx` and have it return the name of the _last_ action. To me it would make more sense for this to do what it says on the tin, and call it with `idx - 1` if needs be - what do you reckon?

Reply:
The problem is: `UndoStack` indexes are state indexes, and not action indexes in `m_macroList`. Because there is an initial state with index 0, the `UndoStack` state index of a given state is always one higher than the index of the action in `m_macroList` that led to this state. It's a bit like this:
<img width="819" alt="Scherm­afbeelding 2024-11-01 om 13 43 07" src="https://github.com/user-attachments/assets/1f4570fc-d2c5-43bd-8b13-e062e05864e4">

I could rename this method to `NotationUndoStack::lastActionNameAtIndex`.

Reply:
Ok got you - thanks.

> I could rename...

Yeah I think this would be clearer - and/or a little comment detailing the above.